material_studies/rally/lib/finance.dart

```diff
@@ -0,0 +1,311 @@
+                    SizedBox(width: 32.0, child: suffix)
+                  ],
+                ),
+              ),
+              Padding(
+                padding: EdgeInsets.only(left: 16, right: 16),
+                child: SizedBox(
+                    height: 1, child: Container(color: Color(0xAA282828))),
+              )
+            ],
+          )),
+    );
+  }
+}
+
+/// Data model for [FinancialEntityCategoryView].
+class FinancialEntityCategoryModel {
+  final Color indicatorColor;
+  final double indicatorFraction;
+  final String title;
+  final String subtitle;
+  final double usdAmount;
+  final Widget suffix;
+
+  const FinancialEntityCategoryModel(
+    this.indicatorColor,
+    this.indicatorFraction,
+    this.title,
+    this.subtitle,
+    this.usdAmount,
+    this.suffix,
+  );
+}
+
+/// Utils for [FinancialEntityCategoryView].
+class FinancialEntityCategoryViews {
+  static FinancialEntityCategoryView fromAccountItem(AccountData model, int i) {
+    return FinancialEntityCategoryView(
+      suffix: Icon(Icons.chevron_right, color: Colors.grey),
+      title: model.name,
+      subtitle: '• • • • • • ' + model.accountNumber.substring(6),
+      indicatorColor: RallyColors.accountColor(i),
+      indicatorFraction: 1.0,
+      amount: model.primaryAmount,
+    );
+  }
+
+  static FinancialEntityCategoryView fromBillItem(BillData model, int i) {
+    return FinancialEntityCategoryView(
+      suffix: Icon(Icons.chevron_right, color: Colors.grey),
+      title: model.name,
+      subtitle: model.dueDate,
+      indicatorColor: RallyColors.billColor(i),
+      indicatorFraction: 1.0,
+      amount: model.primaryAmount,
+    );
+  }
+
+  static FinancialEntityCategoryView fromBudgetItem(
+      BudgetData item, int i, BuildContext context) {
+    return FinancialEntityCategoryView(
+      suffix: Text(' LEFT',
+          style: Theme.of(context)
+              .textTheme
+              .body1
+              .copyWith(color: RallyColors.gray60, fontSize: 10.0)),
+      title: item.name,
+      subtitle: Formatters.usdWithSign.format(item.amountUsed) +
+          ' / ' +
+          Formatters.usdWithSign.format(item.primaryAmount),
+      indicatorColor: RallyColors.budgetColor(i),
+      indicatorFraction: item.amountUsed / item.primaryAmount,
+      amount: item.primaryAmount - item.amountUsed,
+    );
+  }
+
+  static List<FinancialEntityCategoryView> fromAccountDataList(
+      List<AccountData> items) {
+    return List<FinancialEntityCategoryView>.generate(items.length,
+        (i) => FinancialEntityCategoryViews.fromAccountItem(items[i], i));
+  }
+
+  static List<FinancialEntityCategoryView> fromBillDataList(
+      List<BillData> items) {
+    return List<FinancialEntityCategoryView>.generate(items.length,
+        (i) => FinancialEntityCategoryViews.fromBillItem(items[i], i));
+  }
+
+  static List<FinancialEntityCategoryView> fromBudgetItems(
```

Comment:
How are these eventually going to be used? If they're only going to be used by a single Widget, [collection for](https://medium.com/dartlang/announcing-dart-2-3-optimized-for-building-user-interfaces-e84919ca1dff) can probably do everything necessary to keep the code clean.

Reply:
Done